### PR TITLE
[Android/Decoder] Add the decoder sub-plugin for FlatBuffers to AAR @open sesame 12/14 18:33

### PR DESCRIPTION
**Pull request by wooksong**
Fixes https://github.com/nnstreamer/nnstreamer/issues/2928

This patch adds the decoder sub-plugin for FlatBuffers to the Android library (i.e., AAR).

Signed-off-by: Wook Song <wook16.song@samsung.com>

### CODE REVIEW

File: api/android/api/src/main/jni/Android-dec-flatbuf.mk [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#------------------------------------------------------
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := flatbuffers-lib
+LOCAL_SRC_FILES := $(FLATBUF_LIB_PATH)/libflatbuffers.a
+
+include $(PREBUILT_STATIC_LIBRARY)
+
+#------------------------------------------------------
+# tensor-filter sub-plugin for tensorflow-lite
```
Comment (jaeyun-jung) [inline api/android/api/src/main/jni/Android-dec-flatbuf.mk:53]: plz update this line

File: api/android/api/src/main/jni/Android.mk [hunk 2]
```diff
@@ -89,6 +92,11 @@ NNS_SUBPLUGINS += snpe-subplugin
 include $(LOCAL_PATH)/Android-snpe.mk
 endif
 
+ifeq ($(ENABLE_DECODER_FLATBUF),true)
+include $(LOCAL_PATH)/Android-dec-flatbuf.mk
+NNS_SUBPLUGINS += flatbuffers-subplugin
```
Comment (jaeyun-jung) [inline api/android/api/src/main/jni/Android.mk:97]: should add init function for flatbuf plugin. please see below:
https://github.com/nnstreamer/nnstreamer/blob/386b56722dc7bf39885dc6e8cb644b2455e9d8bf/api/android/api/src/main/jni/nnstreamer-native-api.c#L35

File: api/android/api/src/main/jni/Android-dec-flatbuf.mk (hunk 1 above)
Comment (wooksong) [reply api/android/api/src/main/jni/Android-dec-flatbuf.mk:53]: Thanks. Corrected.

File: api/android/api/src/main/jni/Android.mk (hunk 2 above)
Comment (wooksong) [reply api/android/api/src/main/jni/Android.mk:98]: @jaeyun-jung Updated. Thanks.

File: api/android/build-android-lib.sh [hunk 3]
```diff
@@ -341,6 +349,14 @@ if [[ $enable_tflite == "yes" ]]; then
     tar -xJf ./external/tensorflow-lite-$tf_lite_ver.tar.xz -C ./api/src/main/jni
 fi
 
+
+if [[ $enable_decoder_flatbuf == "yes" ]]; then
+    sed -i "s|ENABLE_DECODER_FLATBUF := false|ENABLE_DECODER_FLATBUF := true|" api/src/main/jni/Android.mk
+    sed -i "s|FLATBUF_VER := @FLATBUF_VER@|FLATBUF_VER := ${decoder_flatbuf_ver}|" api/src/main/jni/Android-dec-flatbuf.mk
+    wget --directory-prefix=./external https://raw.githubusercontent.com/nnstreamer/nnstreamer-android-resource/master/external/flatbuffers-${decoder_flatbuf_ver}.tar.xz
```
Comment (gichan-jang) [inline api/android/build-android-lib.sh:356]: How about giving a `-N` option to prevent re-download of the already existing files?

**Review by gichan-jang — APPROVED** [review]
LGTM

File: api/android/api/src/main/jni/Android-dec-flatbuf.mk [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+endif
+
+ifneq ($(SYS_FLATC_VER), $(FLATBUF_VER))
+$(error Found 'flatc' v$(SYS_FLATC_VER), but required v$(FLATBUF_VER))
+endif
+
+FLATBUF_DIR := $(LOCAL_PATH)/flatbuffers
+FLATBUF_INCLUDES := $(FLATBUF_DIR)/include
+GEN_FLATBUF_HEADER := $(shell flatc --cpp -o $(LOCAL_PATH) $(NNSTREAMER_ROOT)/ext/nnstreamer/include/nnstreamer.fbs )
+FLATBUF_HEADER_GEN := $(wildcard $(LOCAL_PATH)/nnstreamer_generated.h2)
```
Comment (gichan-jang) [inline api/android/api/src/main/jni/Android-dec-flatbuf.mk:32]: What is the `nnstreamer_generated.h2`?

File: api/android/build-android-lib.sh (hunk 3 above)
Comment (wooksong) [reply api/android/build-android-lib.sh:358]: @gichan-jang Neither ```-N``` or ```-nc``` could not handle the cases when the downloading is canceled in the middle of it. Therefore, I just added ```rm``` to remove all ```*.tar.gz``` and ```*.tar.xz``` before the downloading gets started.

File: api/android/api/src/main/jni/Android-dec-flatbuf.mk (hunk 4 above)
Comment (wooksong) [reply api/android/api/src/main/jni/Android-dec-flatbuf.mk]: @gichan-jang Corrected this typo. Thanks.

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**